usaspending_api/recipient/delta_models/recipient_lookup.py

```diff
@@ -1,30 +1,583 @@
+        FROM raw.sam_recipient
+        WHERE COALESCE(ultimate_parent_uei, ultimate_parent_unique_ide) IS NOT NULL AND ultimate_parent_legal_enti IS NOT NULL
+    ),
+    latest_tx_parent AS (
+        SELECT
+            4 AS priority,
+            parent_recipient_hash AS recipient_hash,
+            duns_parent_recipient_hash AS duns_recipient_hash,
+            ultimate_parent_legal_enti AS legal_business_name,
+            ultimate_parent_unique_ide AS duns,
+            ultimate_parent_uei AS uei,
+            CONCAT(source, '-parent') AS source,
+            NULL AS address_line_1,
+            NULL AS address_line_2,
+            NULL AS business_types_codes,
+            NULL AS city,
+            NULL AS congressional_district,
+            NULL AS country_code,
+            ultimate_parent_unique_ide AS parent_duns,
+            ultimate_parent_legal_enti AS parent_legal_business_name,
+            ultimate_parent_uei AS parent_uei,
+            NULL AS state,
+            NULL AS zip4,
+            NULL AS zip5,
+            action_date AS update_date,
+            NULL AS alternate_names,
+            ROW_NUMBER() OVER (PARTITION BY parent_recipient_hash ORDER BY action_date DESC, is_fpds, transaction_unique_id) AS row_num
+        FROM temporary_transaction_recipients_view
+        WHERE COALESCE(ultimate_parent_uei, ultimate_parent_unique_ide) IS NOT NULL AND ultimate_parent_legal_enti IS NOT NULL
+    ),
+    latest_duns_sam_no_name AS (
+        SELECT
+            5 AS priority,
+            REGEXP_REPLACE(
+                MD5(UPPER(
+                    CASE WHEN uei IS NOT NULL THEN CONCAT('uei-', uei)
+                    ELSE CONCAT('duns-', awardee_or_recipient_uniqu) END
+                )),
+                '^(\.{{8}})(\.{{4}})(\.{{4}})(\.{{4}})(\.{{12}})$',
+                '\$1-\$2-\$3-\$4-\$5'
+            ) AS recipient_hash,
+            REGEXP_REPLACE(
+                MD5(UPPER(CONCAT('duns-', awardee_or_recipient_uniqu))),
+                '^(\.{{8}})(\.{{4}})(\.{{4}})(\.{{4}})(\.{{12}})$',
+                '\$1-\$2-\$3-\$4-\$5'
+            ) AS duns_recipient_hash,
+            UPPER(legal_business_name) AS legal_business_name,
+            awardee_or_recipient_uniqu AS duns,
+            uei,
+            'sam' AS source,
+            address_line_1,
+            address_line_2,
+            business_types_codes,
+            city,
+            congressional_district,
+            country_code,
+            ultimate_parent_unique_ide AS parent_duns,
+            UPPER(ultimate_parent_legal_enti) AS parent_legal_business_name,
+            ultimate_parent_uei AS parent_uei,
+            state,
+            zip4,
+            zip AS zip5,
+            update_date,
+            NULL AS alternate_names,
+            ROW_NUMBER() OVER (PARTITION BY uei, awardee_or_recipient_uniqu ORDER BY update_date DESC NULLS LAST) AS row_num
```

Comment:
Same thing here, since one could be null, probably want to partition by a coalesce of both:

`PARTITION BY COALESCE(uei, awardee_or_recipient_uniqu) ORDER BY update_date DESC NULLS LAST) AS row_num`